### ISSUE TITLE
Cannot simultaneously set x-axis and x-axis rangeslider limits

### ISSUE THREAD

**Issue by cczhu** [plotly/plotly.py#2695]
I'm plotting timeseries data that includes an x-axis rangeslider within a Dash app, and there are some circumstances where I would like to change the x-axis limits of the plot, but also set the x-axis rangeslider to a range within these limits (for example set the limits to `['2015-05-01', '2017-01-01']` and the rangeslider to `['2015-07-01', '2015-08-01']`). [The documentation on the x-axis](https://plotly.com/python/reference/layout/xaxis/#layout-xaxis-rangeslider) makes it sound as if setting `xaxis_rangeslider_range` should do the trick, however I'm having some trouble using this.

I'm using 

```
plotly==4.9.0
```

inside a Jupyter Notebook environment within Windows Subsystem for Linux with Ubuntu 18.04.

Small reproducible test case

```python
import plotly.express as px
import pandas as pd

df = pd.read_csv('https://raw.githubusercontent.com/plotly/datasets/master/finance-charts-apple.csv')

fig = px.line(df, x='Date', y='AAPL.High', title='Time Series with Range Slider and Selectors')

fig.update_xaxes(
    rangeslider_visible=True,
    range=['2015-05-01', '2017-01-01'],
)
fig.show()
```

This gives me:

![image](https://user-images.githubusercontent.com/442891/89720519-efd61180-d9a0-11ea-89aa-143f61bf4ef2.png)

If I instead do

```python
fig.update_xaxes(
    rangeslider_visible=True,
    range=['2015-05-01', '2017-01-01'],
    rangeslider_range=['2015-07-01', '2015-08-01']
)
fig.show()
```

![image](https://user-images.githubusercontent.com/442891/89720529-0d0ae000-d9a1-11ea-9a6b-a98c09febada.png)

The x-axis limits **and** rangeslider initial limits are set to `['2015-05-01', '2017-01-01']`, regardless of what `rangeslider_range` is set to.

Is this expected behaviour? If so, is there a way to set the initial rangeslider limits as well as the x-axis limits without removing some of the data being plotted?

**Comment by nicolaskruchten** [comment]
Thanks for the very nice test cases :)

I believe this is the expected behaviour, yes. The naming is a bit confusing here, but the selected portion of the rangeslider is in fact the xaxis range, which is why when you interact with it, the main plot changes. When setting only the range of the axis, the rangeslider therefore shows a subset of the chart as "selected".

As a result, the `rangeslider_range` parameter can't be used to set the range to something smaller than the xaxis `range`, as you're trying to do here, but can be set to be bigger:

```python
import plotly.express as px
import pandas as pd

df = pd.read_csv('https://raw.githubusercontent.com/plotly/datasets/master/finance-charts-apple.csv')

fig = px.line(df, x='Date', y='AAPL.High', title='Time Series with Range Slider and Selectors')

fig.update_xaxes(
    rangeslider_visible=True,
    range=['2015-05-01', '2015-06-01'],
    rangeslider_range=['2015-03-01', '2015-08-01']
)
fig.show()
```

![image](https://user-images.githubusercontent.com/203523/89790850-8d8c2680-daf0-11ea-8f0b-c8794f6888c3.png)


So basically I think that swapping your values for `range` and `rangeslider_range` will get you what you're looking for, although I might be misinterpreting!

**Comment by cczhu** [comment]
Ahh yes, that's the behaviour I'm looking for! Thank you for clarifying the functionality of the two keywords - I did find the documentation to be a bit confusing. Maybe pointing out that `rangeslider_range` sets the maximum extent of the range, and `range` its initial extent, would clarify?

Thanks so much for your help! Closing the issue, since the behaviour is expected.
